### PR TITLE
Make pam.d directory configurable

### PR DESCRIPTION
**Pull request by yifanjiang**
Change according to the discussion in https://github.com/neutrinolabs/xrdp/pull/2547 .

### CODE REVIEW

File: configure.ac
Comment (metalefty) [inline configure.ac]: I think the default should location be `$sysconfdir/etc` rather than `/etc` when pamconfdir is not specified. Otherwise, this change will change current behaviour.

Comment (metalefty) [reply configure.ac]: Example: https://cgit.freedesktop.org/systemd/systemd/commit/?id=5c390a4ae0d383b2003074ed011d47876c7e630c

**Comment by matt335672** [comment]
Hi @yifanjiang 

I'm out of time this week to have a good look at this, but after a quick glance this indeed looks like I was thinking of. This will definitely be useful to SuSE and any other distros that don't use `/etc/pam.d`.

A quick question- is there a good reason why you didn't add `/pam.d` to the pam config dir?

Speak soon.

**Comment by yifanjiang** [comment]
> A quick question- is there a good reason why you didn't add `/pam.d` to the pam config dir?

Hi @matt335672 , thanks for looking into this. The reason was to make it a little more flexible for downstream packaging. While I took a second look, keeping a consistent lexical style with socketdir and sesmanruntimedir might be a better choice. So I update through the changes.

**Review by metalefty — APPROVED** [review]
I'm fine with this now. Waiting for @matt335672 's approval.

File: sesman/verify_user_pam.c
Comment (matt335672) [inline sesman/verify_user_pam.c:207]: Before line 207, can you add:--

```c
#ifdef __LINUX_PAM__
            /* /usr/lib/pam.d is hardcoded into Linux-PAM */
            g_file_exist("/usr/lib/pam.d/xrdp-sesman")) ||
#endif
#ifdef OPENPAM_VERSION
            /* /usr/local/etc/pam.d is hardcoded into OpenPAM */
            g_file_exist("/usr/local/etc/pam.d/xrdp-sesman")) ||
#endif
```

**Review by matt335672 — COMMENTED** [review]
I was all ready to approve this, then I read this [useful comment](https://github.com/neutrinolabs/xrdp/pull/2547#issuecomment-1430897118) from @thkukuk

I suggest the following amendment.

@yifanjiang - I apologise for messing you about like this. However, what we've now got something that should be able to support the use-cases mentioned by @thkuk

File: sesman/verify_user_pam.c
Comment (yifanjiang) [reply sesman/verify_user_pam.c:207]: @matt335672 That's completely fine, I am happy there could be a better connection between xrdp and pam. Change already committed.

**Comment by matt335672** [comment]
Thanks for the quick edits @yifanjiang 

Again, I apologise for the confusion.

**Comment by matt335672** [comment]
@metalefty - I think this should be backported to v0.9. Do you agree?

**Comment by metalefty** [comment]
@matt335672 Yes, it is definitely useful.

**Pull request merged**
